pkg/controllers/subnet/subnet_controller.go

```diff
@@ -0,0 +1,272 @@
+			return ResultRequeue, err
+		}
+		if err := r.updateSubnetStatus(obj); err != nil {
+			log.Error(err, "update subnet status failed, would retry exponentially", "subnet", req.NamespacedName)
+			updateFail(r, &ctx, obj, &err)
+			return ResultRequeue, err
+		}
+		updateSuccess(r, &ctx, obj)
+	} else {
+		if controllerutil.ContainsFinalizer(obj, servicecommon.FinalizerName) {
+			metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerDeleteTotal, MetricResTypeSubnet)
+			if err := r.Service.DeleteSubnet(obj, &vpcInfo); err != nil {
+				log.Error(err, "deletion failed, would retry exponentially", "subnet", req.NamespacedName)
+				deleteFail(r, &ctx, obj, &err)
+				return ResultRequeue, err
+			}
+			controllerutil.RemoveFinalizer(obj, servicecommon.FinalizerName)
+			if err := r.Client.Update(ctx, obj); err != nil {
+				log.Error(err, "deletion failed, would retry exponentially", "subnet", req.NamespacedName)
+				deleteFail(r, &ctx, obj, &err)
+				return ResultRequeue, err
+			}
+			log.V(1).Info("removed finalizer", "subnet", req.NamespacedName)
+			deleteSuccess(r, &ctx, obj)
+		} else {
+			log.Info("finalizers cannot be recognized", "subnet", req.NamespacedName)
+		}
+	}
+	return ctrl.Result{}, nil
+}
+
+func (r *SubnetReconciler) updateSubnetStatus(obj *v1alpha1.Subnet) error {
+	nsxSubnet := r.Service.SubnetStore.GetByKey(string(obj.GetUID()))
+	obj.Status.IPAddresses = obj.Status.IPAddresses[:0]
+	statusList, err := r.Service.GetSubnetStatus(nsxSubnet)
+	if err != nil {
+		return err
+	}
+	for _, status := range statusList {
+		obj.Status.IPAddresses = append(obj.Status.IPAddresses, *status.NetworkAddress)
+	}
+	obj.Status.NSXResourcePath = *nsxSubnet.Path
+	if len(obj.OwnerReferences) > 0 {
+		// Updates SubnetSet status if Subnet belongs to one SubnetSet.
+		subnetset := &v1alpha1.SubnetSet{}
+		key := types.NamespacedName{
+			Namespace: obj.Namespace,
+			Name:      obj.OwnerReferences[0].Name,
+		}
+		if err := r.Client.Get(context.Background(), key, subnetset); err != nil {
+			return err
+		}
+		newSubnetInfo := v1alpha1.SubnetInfo{
+			NSXResourcePath: obj.Status.NSXResourcePath,
+			IPAddresses:     obj.Status.IPAddresses,
+		}
+		exist := false
+		for i, existingSubnetInfo := range subnetset.Status.Subnets {
+			if existingSubnetInfo.NSXResourcePath == obj.Status.NSXResourcePath {
+				subnetset.Status.Subnets[i] = newSubnetInfo
+				exist = true
+				break
+			}
+		}
+		if !exist {
+			subnetset.Status.Subnets = append(subnetset.Status.Subnets, newSubnetInfo)
+		}
+		if err := r.Client.Status().Update(context.Background(), subnetset); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *SubnetReconciler) setSubnetReadyStatusTrue(ctx *context.Context, subnet *v1alpha1.Subnet) {
+	newConditions := []v1alpha1.Condition{
+		{
+			Type:    v1alpha1.Ready,
+			Status:  v1.ConditionTrue,
+			Message: "NSX Subnet has been successfully created/updated",
+			Reason:  "NSX API returned 200 response code for PATCH",
```

Comment:
I checked K8s common reasons, they use brief phase  "Reason:  "InstanceDeletionInProgress",
		Message: "CustomResource deletion is in progress",", for example  https://github.com/kubernetes/kubernetes/blob/2c6c4566eff972d6c1320b5f8ad795f88c822d09/staging/src/k8s.io/apiextensions-apiserver/pkg/controller/finalizer/crd_finalizer.go#L129 and 
https://github.com/kubernetes/kubernetes/blob/2c6c4566eff972d6c1320b5f8ad795f88c822d09/pkg/controller/controller_utils.go#L414
Could we also follow the same convention?
Also I think we do not need to say "NSX API return 200" , just say "SubnetCreated"

Reply:
Thanks, done.